src/js/app.js

```diff
@@ -0,0 +1,151 @@
+import { calculatorInitValue, calculatorSymbols, calculatorDigitMaxLength } from './constants.js';
+import { calculate } from './calculator.js';
+
+class App {
+  /** @type {HTMLDivElement} */
+  #element;
+  /** @type {HTMLHeadingElement} */
+  #totalElement;
+  /** @type {boolean} */
+  #isCalculateAfter = false;
+
+  /**
+   * @param {HTMLDivElement} element
+   */
+  constructor(element) {
+    this.#element = element;
+    this.#render();
+    this.#mounted();
+  }
+
+  /**
+   * @param {string} value
+   */
+  calculatorInput(value) {
+    const total = this.#totalElement.textContent;
+    const regExp = /([0-9]+)/g;
+    const totalLastNumbers = total.match(regExp).pop();
+    const isLastStringContainsSymbol = calculatorSymbols.includes(total.slice(-1));
+
+    // 심볼을 입력한 경우
+    if (calculatorSymbols.includes(value)) {
+      this.#totalElement.textContent = isLastStringContainsSymbol
+        ? total.substr(0, total.length - 1) + value
+        : total + value;
+      return;
+    }
+
+    // 번호를 입력했지만 번호가 최대 자리수를 넘치지 않은 경우
+    if (totalLastNumbers.length < calculatorDigitMaxLength || isLastStringContainsSymbol) {
+      this.#totalElement.textContent = total === calculatorInitValue ? value : total + value;
+      return;
+    } else {
+      throw new Error(`번호를 ${calculatorDigitMaxLength}보다 길게 작성할 수 없습니다.`);
+    }
+  }
+
+  /**
+   * @param {MouseEvent} e
+   */
+  #handleDigitsClick = (e) => {
+    /**
+     * @typedef {object} DestructuredEvent
+     * @property {Element} target
+     */
+    /** @type {DestructuredEvent} */
+    const { target } = e;
+
+    if (target.classList.contains('digit') && this.#isCalculateAfter === false) {
```

Comment:
```suggestion
    if (target.classList.contains('digit') && !this.#isCalculateAfter) {
```

이런 식으로 축약해서 표현할 수 있을거 같네용

Reply:
false를 사용하는 것보다 !를 사용하는것이 더 일반적일까요?
개인적으로 저는 !가 눈에 잘 들어오지 않는 것 같아서 false 형태로 작성을 했습니다 ㅜㅜ